src/bitset.hpp

```diff
@@ -354,6 +411,36 @@ TEST_CASE("Bitset") {
   // "Lift" a bitset.
   CHECK_EQ(Remap(Bitset("11"), {0, std::nullopt, 1}), Bitset("101"));
 }
+
+// See simple diagram at:
+// https://user-images.githubusercontent.com/31897211/136752835-2ad4649b-dc14-4c80-8174-d0567e849cf8.gif
+// See tree diagram at:
+// https://user-images.githubusercontent.com/31897211/136849710-de0dcbe3-dc2b-42b7-b3de-dd9b1a60aaf4.gif
+TEST_CASE("Bitset: Nearest Neighbor Interchange") {
+  // Clades for NNI.
+  Bitset X("100");
+  Bitset Y("010");
+  Bitset Z("001");
+  // Initial Child and Parent.
+  Bitset parent_in = Bitset::Subsplit(X, Y | Z);
+  Bitset child_in = Bitset::Subsplit(Y, Z);
+  // Correct Solutions.
+  Bitset correct_parent_xy = Bitset::Subsplit(Y, X | Z);
+  Bitset correct_child_xy = Bitset::Subsplit(X, Z);
+  Bitset correct_parent_xz = Bitset::Subsplit(Z, Y | X);
+  Bitset correct_child_xz = Bitset::Subsplit(Y, X);
+  std::cout << "Correct Solutions" << std::endl;
+
+  // Swap X and Y
+  auto [parent_xy, child_xy] = Bitset::SubsplitNNI(parent_in, child_in, 0);
+  CHECK_EQ(correct_parent_xy, parent_xy);
+  CHECK_EQ(correct_child_xy, child_xy);
+  // Swap X and Z
+  auto [parent_xz, child_xz] = Bitset::SubsplitNNI(parent_in, child_in, 1);
+  CHECK_EQ(correct_parent_xz, parent_xz);
+  CHECK_EQ(correct_child_xz, child_xz);
+}
```

Comment:
If we want to expose the 4-argument SubsplitNNI to this then let's test it, including one that fails.